crates/driver/src/infra/pod/api.rs

```diff
@@ -0,0 +1,257 @@
+use {
+    crate::domain::eth,
+    alloy::{
+        network::{EthereumWallet, TxSigner},
+        primitives::Log,
+        signers::{Signature, Signer, aws::AwsSigner, k256, local::LocalSigner},
+        sol,
+        transports::{RpcError, TransportErrorKind},
+    },
+    ethcontract::{
+        Account::{self, Kms},
+        Address,
+        U256,
+    },
+    pod_sdk::{
+        alloy_primitives,
+        alloy_sol_types::SolEvent,
+        auctions::client::AuctionClient,
+        provider::{PodProvider, PodProviderBuilder},
+    },
+    pod_types::LogFilterBuilder,
+    serde::Deserialize,
+    serde_with::serde_as,
+    std::time::{Duration, UNIX_EPOCH},
+    thiserror::Error,
+    url::Url,
+};
+
+sol! {
```

Comment:
Would be nice if this would go the usual flow of defining an ABI file in `/contracts/artifacts`